server/routes.go

```diff
@@ -1529,6 +1530,24 @@ func (s *Server) ChatHandler(c *gin.Context) {
 		defer close(ch)
 		var sb strings.Builder
 		var toolCallIndex int = 0
+		var templateToolToken string
+		var tmpl *gotmpl.Template
+		if len(req.Tools) > 0 {
+			var ok bool
+			templateToolToken, ok = ToolToken(m.Template.Template)
+			if !ok {
+				slog.Debug("no tool token found")
+			}
+			tmpl, ok = ToolTemplate(m)
+			if !ok {
+				slog.Debug("no tool template found")
+			}
+		}
+
+		checkToolCall := false
```

Comment:
Can we rename or ideally eliminate this variable? I think it's purpose makes sense (whether or not we should attempt to parse tools) but it would be even more awesome if the control flow below could handle that vs having more state (it makes these handlers harder to edit later)